src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/FileProvXAttrBufferClusterj.java

```diff
@@ -0,0 +1,122 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hops.metadata.ndb.dalimpl.hdfs;
+
+import com.mysql.clusterj.annotation.Column;
+import com.mysql.clusterj.annotation.PersistenceCapable;
+import com.mysql.clusterj.annotation.PrimaryKey;
+import io.hops.exception.StorageException;
+import io.hops.metadata.hdfs.TablesDef;
+import io.hops.metadata.hdfs.dal.FileProvXAttrBufferDataAccess;
+import io.hops.metadata.hdfs.entity.FileProvXAttrBufferEntry;
+import io.hops.metadata.ndb.ClusterjConnector;
+import io.hops.metadata.ndb.wrapper.HopsSession;
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class FileProvXAttrBufferClusterj implements TablesDef.FileProvXAttrBufferTableDef,
+  FileProvXAttrBufferDataAccess<FileProvXAttrBufferEntry> {
+
+  private ClusterjConnector connector = ClusterjConnector.getInstance();
+
+  @PersistenceCapable(table = TABLE_NAME)
+  public interface FileProvXAttrBufferEntryDto {
+
+    @PrimaryKey
+    @Column(name = INODE_ID)
+    long getINodeId();
+
+    void setINodeId(long inodeId);
+
+    @PrimaryKey
+    @Column(name = NAMESPACE)
+    byte getNamespace();
+
+    void setNamespace(byte id);
+
+    @PrimaryKey
+    @Column(name = NAME)
+    String getName();
+
+    void setName(String name);
+    
+    @PrimaryKey
+    @Column(name = INODE_LOGICAL_TIME)
+    int getINodeLogicalTime();
+
+    void setINodeLogicalTime(int inodeLogicalTime);
+
+    @Column(name = VALUE)
+    String getValue();
+
+    void setValue(String value);
+  }
+
+  private FileProvXAttrBufferEntryDto createPersistable(FileProvXAttrBufferEntry logEntry) throws StorageException {
+    HopsSession session = connector.obtainSession();
+    FileProvXAttrBufferEntryDto dto = session.newInstance(FileProvXAttrBufferEntryDto.class);
+    dto.setINodeId(logEntry.getInodeId());
+    dto.setNamespace(logEntry.getNamespace());
+    dto.setName(logEntry.getName());
+    dto.setINodeLogicalTime(logEntry.getINodeLogicalTime());
+    dto.setValue(logEntry.getValue());
+    return dto;
+  }
+
+  private Collection<FileProvXAttrBufferEntry> createCollection(Collection<FileProvXAttrBufferEntryDto> collection) {
+    ArrayList<FileProvXAttrBufferEntry> list = new ArrayList<>(collection.size());
+    for (FileProvXAttrBufferEntryDto dto : collection) {
+      list.add(createProvenanceLogEntry(dto));
+    }
+    return list;
+  }
+
+  private FileProvXAttrBufferEntry createProvenanceLogEntry(FileProvXAttrBufferEntryDto dto) {
+    return new FileProvXAttrBufferEntry(
+      dto.getINodeId(),
+      dto.getNamespace(),
+      dto.getName(),
+      dto.getINodeLogicalTime(),
+      dto.getValue());
+  }
```

Comment:
Is there a plane to use these functions in the future? Write now they are not usefull as we only write to the database and never read from it.

Reply:
Fixed, removed createCollection and createProvenanceLogEntry - it was dead code